French version

jean-gui opened the pull request:
Brings French to Kanzi. Related to #266

jean-gui committed:
Added French

Xaelias commented on alexa.py:
```diff
@@ -44,8 +44,13 @@
   app.config['ASK_APPLICATION_ID'] = os.getenv('MEDIA_CENTER_SKILL_ID')
 
 LANGUAGE = config.get('global', 'language')
```
Not sure what this get is. If you can provide a default value there, do it.
Otherwise, do something like:
```
if LANGUAGE == 'None':
  LANGUAGE == 'en'
```
Which in turns simplify the if/then/else under it.

jean-gui replied:
The get retrieves the language value from kodi.config. No idea if it's possible to provide a default value, so I implemented your suggestion.

m0ngr31 replied:
I'll look into this. We should be able to default it to english.

Xaelias commented on templates.fr.yaml:
```diff
@@ -0,0 +1,449 @@
+
+hosted_config_missing: "Je n'ai pas trouvé toutes les informations de connexion dont j'ai besoin. Veuillez mettre à jour vos informations sur app.media-dash-center.io"
+
+not_logged_in: "Veuillez lier votre compte pour utiliser cette skill"
```
un skill non?

jean-gui replied:
Dans le menu de https://developer.amazon.com/fr/alexa-skills-kit il y a "Raccourci : Commencez une Skill maintenant" donc je suppose que c'est bien féminin :p

Xaelias replied:
J'ai toujours dit un skill. Vient peut-être de mon background de gamer... A priori en Allemand ils disent Die et Das (donc féminin et neutre). En Anglais c'est neutre.
Après je dis une gameboy et beaucoup de gens disent un gameboy du coup...
Perso je vote un skill. Mais j'utiliserai jamais la version français du coup...

jean-gui replied:
Vu que le site officiel utilise le féminin, je pense que c'est mieux qu'on utilise le féminin aussi.

bisca330 replied:
Bonjour,

Peut-être l'avez-vous déjà vu, mais sait-on jamais : https://www.alexafrance.fr/25/05/2018/alexa-faut-il-dire-un-skill-ou-une-skill/

Cdlt,

Micky

Xaelias commented on templates.fr.yaml:
```diff
@@ -0,0 +1,449 @@
+
+volume_set: "Volume réglé sur {{ num }}"
+
+help: "Vous pouvez me demander s'il y a des nouvelles séries, de lire un film, une sérier ou un artiste, contrôler la lecture des médias et plus encore. Par exemple, {{ example }}. J'ai envoyé plus d'exemple à votre application Alexa."
```
plus d'exemples

Xaelias commented on templates.fr.yaml:
```diff
@@ -0,0 +1,449 @@
+
+looking_for_show: "Recherche de nouveaux épisodes de {{ heard_show }}"
+
+current_playing_item: "Élement en cours de lecture"
```
Élément

Xaelias commented on templates.fr.yaml:
```diff
@@ -0,0 +1,449 @@
+
+newly_added_movies_genre: "Films de {{ genre }} récemment ajoutés"
+
+last_unwatched: "Lecture de l'épisode non vu suivant de la dernière série regardée"
```
C'est correct. Mais je me verrai pas utiliser ça au quotidien.
Un truc du style "Lecture du prochain épisode de la ..." serait un peu plus fluide je pense.

jean-gui replied:
C'est vrai que c'est lourd. Je vais le changer. Note que templates.fr.yaml ne contient que les messages renvoyés par le serveur, pas les phrases qu'on prononce.

Pour ce qu'on prononce, j'ai mis dans utterances.fr.txt:
WatchNextEpisode (joue/démarre/regarde/lis) l'épisode suivant de {Show}

jean-gui replied:
J'ai aussi modifié WatchNextEpisode pour permettre de dire 'le prochain épisode'

Xaelias reviewed (commented):
J'ai juste fait une lecture très rapide. J'essayerai de faire un truc un peu plus en profondeur quand j'aurai du temps.
Merci en tous cas :-)

jean-gui committed:
Simplify template selection from language config

jean-gui committed:
Fixed typos

jean-gui committed:
Rephrase last_unwatched and WatchNextEpisode

bisca330 commented:
Bonjour,

Merci pour le travail de "francisation" de cette skill.

Je vais tester ça chez moi au plus vite pour voir ce que ça donne.

Bien cordialement,

Micky

bisca330 commented:
Encore moi, 

Je viens de décortiquer un peu le fichier "utterances.fr.txt" et je ne trouve pas de les commandes "pause" ou "stop". Est-ce un oubli ou est-ce que quelque chose m’échappe ?

Cdlt,

Micky

jean-gui commented:
@bisca330 Effectivement il n'y pas pause/stop, mais ils n'ont pas l'air d'être non plus dans la version anglaise... J'avoue que j'utilise assez peu les phrases autres que celles pour démarrer la lecture car de toute façon le dot n'entend plus rien dès qu'il y a le son de la télé :-/

bisca330 commented:
Merci pour l'info sur le lien sur lequel pointer. J'ai essayé de déployer la version anglaise mais j'ai une erreur sur le dernier script qui déploie kanzi et qui me dit que ma configuration kodi n'est pas bonne. Faut que je regarde ça de plus près.

Sinon pour être précis, et l'information vient d'ici, Alexa arrête l'écoute dès qu'un média se lance (vidéo ou musique). Par contre, tant qu'aucun média n'est lancé Alexa est en écoute et on n'est normalement pas obligé de dire la phrase d'invocation avant de demander une commande, on peut donc naviguer dans kodi sans dire cette phrase. Mais ça je ne pourrais le vérifier que lorsque j'aurais réussi à déployer kanzi.

Cdlt,

Micky

jean-gui commented:
Ce que je voulais dire c'est que le dot ne reconnait plus rien dès qu'il y a du son à côté, à moins de lui gueuler dessus ;)

bisca330 commented:
Moi ce qu'on m'a dit c'est que le dot est désactivé dès qu'on demande la lecture d'un média. Il faudrait donc prononcer la phrase d'activation pour réactiver Alexa... Mais en effet avec un home cinéma un peu puissant elle risque d'être sourde dingue 😀

bisca330 commented:
Bonjour,

J'ai ouvert un problème pour les commandes pause et stop qui ne semblent pas présentes dans Kanzi. La réponse est ici : https://github.com/m0ngr31/kanzi/issues/269# Pour résumer elles sont bien présentes dans une bibliothèque spécifique et donc n'apparaissent pas explicitement dans les commandes existantes. Par contre je ne sais pas trop de quelle bibliothèque il est fait mention dans la réponse que j'ai eu et s'il faut y modifier quelque chose pour que cela fonctionne en français.

Jean-gui, as-tu moyen de voir ça ? Tu as l'air de bien toucher ta bille sur le sujet.

Cdlt,

Micky

jingai commented:
I was referring to the [standard built-in intents](https://developer.amazon.com/docs/custom-skills/built-in-intent-library.html#standard-built-in-intents).

There should be nothing you have to do in the language translation for this, though.

bisca330 commented:
Jean-gui,

Tu m'expliquais plus haut que je devais utiliser lexigram version française et modifier le fichier action.ts. Peux-tu me dire comment installer lexigram version fançaise stp ?

J'ai installé la version original en utilisant la commande "npm install -g lexigram-cli" trouvé sur le site de l'auteur https://lexigr.am/docs/getting-started.html. Est-ce que je dois procéder avec le même style de commande ?

En fait ce qui manque c'est un mode opératoire précis pour l'installation de la skill. Le site de l'auteur reste correct mais il manque des précisions, comme par exemple installer nvm sur un système Windows 32 bits alors que l'installation par défaut est prévue pour le 64 bits... ça c'est noté nul part...). N'étant pas des plus doués avec la langue de Shakespeare, n'y connaissant rien à AWS et n'étant pas développeur) j'ai un peu de mal à raccrocher les wagons. Qui plus est je découvre GitHub.

Donc dans un premier temps saurais-tu me dire comment installer lexigram-cli version Molière et au cas où aurais-tu une vague procédure pour l'installation de cette skill en français ?

Cdlt,

Micky

jean-gui commented:
@bisca330 plutôt que de polluer cette PR, j'ai créé une issue sur mon repo perso : https://github.com/jean-gui/lexigram-cli/issues/1 pour en discuter

jean-gui commented:
@m0ngr31 any idea when we can hope to get this merged?

m0ngr31 commented:
I'll try and get everything in place this weekend.

m0ngr31 committed:
Update alexa.py

Changed the way the kodi-voice library worked. It'll always provide a default of english now.

m0ngr31 committed:
Update requirements.txt

New update to the voice library

m0ngr31 reviewed (approved):
I can't read the French comments, so I'm just going to have to assume that everything works. Everything looks good as far as I can tell.

jean-gui commented:
Thanks @m0ngr31. French comments were mostly about how to install the French version

m0ngr31 commented:
Cool. The cli is updated, so you should be able to run it from that
